Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/config/PortalMenuConfig.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * #%L
+ * This file is part of "Apromore Core".
+ * %%
+ * Copyright (C) 2018 - 2021 Apromore Pty Ltd.
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>.
+ * #L%
+ */
+package org.apromore.portal.config;
+
+import org.apromore.plugin.portal.PortalLoggerFactory;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.core.io.Resource;
+import org.json.JSONObject;
+import org.json.JSONArray;
+
+import org.slf4j.Logger;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.Getter;
+
+import org.apromore.plugin.portal.PortalLoggerFactory;
+
+@Component("menuConfig")
+public class PortalMenuConfig {
+
+	private static final Logger LOGGER = PortalLoggerFactory.getLogger(PortalMenuConfig.class);
+
+	private JSONArray menuTree;
+	private Resource menuConfigUrl;
+
+	@Getter
+	private List<String> menuGroups;
+
+	@Getter
+	private Map<String, List<String>> menuiItemMap;
+
+	public PortalMenuConfig(@Value("${portal.menu.config.url}") Resource menuConfigUrl) {
+		this.menuConfigUrl = menuConfigUrl;
+	}
+
+	public void load() {
+		menuGroups = new ArrayList<String>();
+		menuiItemMap  = new HashMap<String, List<String>>();
+
+		try {
+			InputStream is = menuConfigUrl.getInputStream();
+			String jsonString = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+			JSONObject root = new JSONObject(jsonString);
+			menuTree = root.getJSONArray("groups");
+			for (int i = 0; i < menuTree.length(); i++) {
+				JSONObject group = menuTree.getJSONObject(i);
+				String groupId = group.getString("id");
+				JSONArray items = group.getJSONArray("items");
+				menuGroups.add(groupId);
+				List<String> menuItems = new ArrayList<String>();
+				for (int j = 0; j < items.length(); j++) {
+					JSONObject item = items.getJSONObject(j);
+					String itemId = item.getString("id");
+					menuItems.add(itemId);
+				}
+				menuiItemMap.put(groupId, menuItems);
+			}
+		} catch (Exception e) {
+			LOGGER.info("Failed to load menu definition");
+		}
```

Comment:
The exception handling here is not complete. A question here is what is the behavior of the software when different exceptions occur in loading menu config: missing config file, IO problem, missing menu items, wrong value type, wrong format. Some are more serious than the others. 
For example: 
- Missing the menu config file is serious because it can't go forward (nothing can be shown). In this case, the program should inform the user about the error to fix and then stop.
- Missing a menu item, wrong value, etc. is less serious: the program should continue showing other valid menu items, but showing a notification for the user to know.
- Other individual issues: it should be able to continue.

For each type of exceptions above, throw a type of exception for the calling program to know and handle (here is the Portal MainController). Preferred unchecked exceptions rather than checked exception to keep the API clean.